api/src/main/java/org/openmrs/api/InvalidIdentifierFormatException.java

```diff
@@ -15,32 +15,38 @@
 	
 	private static final long serialVersionUID = 1L;
 	
-	private String format;
+	private final String format;
 	
 	public InvalidIdentifierFormatException() {
+		format = null;
+	}
+
+	public InvalidIdentifierFormatException(String message, String format) {
+		super(message);
+		this.format = format;
 	}
 	
 	public InvalidIdentifierFormatException(String message) {
 		super(message);
+		format = null;
 	}
 	
 	public InvalidIdentifierFormatException(String message, PatientIdentifier identifier) {
 		super(message, identifier);
+		format = null;
 	}
 	
 	public InvalidIdentifierFormatException(String message, Throwable cause) {
 		super(message, cause);
+		format = null;
 	}
 	
 	public InvalidIdentifierFormatException(Throwable cause) {
 		super(cause);
+		format = null;
 	}
 	
 	public String getFormat() {
 		return format;
 	}
-	
-	public void setFormat(String format) {
```

Comment:
Can you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Deprecation

Reply:
I will look into it. Thank you for your time!